src/index.ts

```diff
@@ -95,8 +95,8 @@ const typescript: PluginImpl<RPT2Options> = (options) =>
 			verbosity: VerbosityLevel.Warning,
 			clean: false,
 			cacheRoot: findCacheDir({ name: "rollup-plugin-typescript2" }),
-			include: ["*.ts+(|x)", "**/*.ts+(|x)"],
```

Comment:
I think we need to modify [the TS emit output conversion](https://github.com/ezolenko/rollup-plugin-typescript2/blob/68017aeb9eb81f9473bfe037ece52a837c1986b5/src/tscache.ts#L38) to also handle `.d.cts`, `.d.mts` and their respective mappings. I _assume_ that TS would take a CTS file and output `.cjs`, `d.cts`, and `d.cts.map` files, which we would not handle correctly right now.

We should also definitely have integration tests for CTS / MTS, especially to validate that assumption above^

Reply:
Yeah it would fix #448 too

Reply:
Oh I didn't even think about that, but it would indeed fix #448. There is a slight problem there with declarations though; Rollup cannot convert them between `.d.cts` and `.d.mts`, so they are limited to the user's original module format. I'll detail that more in the issue itself.

Reply:
For the emit output conversion, I looked up the source code and indeed `OutputFile` now has some new extensions:
- [`.mjs` + `.cjs`](https://github.com/microsoft/TypeScript/blob/d8585688dd1bc8d82b7b5daab9af83ae1e3de197/src/compiler/emitter.ts#L576)
- [`.d.cts`, `.d.mts`](https://github.com/microsoft/TypeScript/blob/d8585688dd1bc8d82b7b5daab9af83ae1e3de197/src/compiler/utilities.ts#L6229)
- `.map` files of those are just [appended](https://github.com/microsoft/TypeScript/blob/d8585688dd1bc8d82b7b5daab9af83ae1e3de197/src/compiler/emitter.ts#L546) to the existing extension from above

I imagine that with `module` set to output ESM (for Rollup), we'd only see the `.mjs` variants, but testing can confirm that